File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/ui/screens/members/MemberListScreen.kt
package com.gym.gymapp.ui.screens.members

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Search
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.LocalTextStyle
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.material3.pulltorefresh.PullToRefreshBox
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.gym.gymapp.ui.components.MemberItem
import com.gym.gymapp.ui.viewmodels.MemberListViewModel
import org.koin.compose.viewmodel.koinViewModel

@OptIn(ExperimentalMaterial3Api::class)

@Composable
fun MemberListScreen(
    onBack: () -> Unit,
    onMemberClick: (String) -> Unit,
    onAddMember: () -> Unit,
    viewModel: MemberListViewModel = koinViewModel()
) {
    LaunchedEffect(Unit) {
        viewModel.refresh()
    }

    BoxWithConstraints(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
    ) {
        val isTablet = maxWidth > 600.dp

        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 16.dp)
        ) {

            // Header
            Spacer(Modifier.height(20.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Text(
                    "Members",
                    color = MaterialTheme.colorScheme.onBackground,
                    fontSize = if (isTablet) 24.sp else 18.sp,
                    fontWeight = FontWeight.Black
                )
            }

            Spacer(Modifier.height(16.dp))

            // Search Bar & Filter
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = 8.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                TextField(
                    value = viewModel.uiState.searchQuery,
                    onValueChange = { viewModel.onSearchQueryChange(it) },
                    modifier = Modifier
                        .weight(1f)
                        .height(52.dp),
                    placeholder = {
                        Text(
                            "Search...",
                            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f),
                            fontSize = 13.sp
                        )
                    },
                    leadingIcon = {
                        Icon(
                            Icons.Default.Search,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f),
                            modifier = Modifier.size(18.dp)
                        )
                    },
                    colors = TextFieldDefaults.colors(
                        focusedContainerColor = MaterialTheme.colorScheme.surface,
                        unfocusedContainerColor = MaterialTheme.colorScheme.surface,
                        focusedIndicatorColor = Color.Transparent,
                        unfocusedIndicatorColor = Color.Transparent,
                        cursorColor = MaterialTheme.colorScheme.primary,
                        focusedTextColor = MaterialTheme.colorScheme.onSurface,
                        unfocusedTextColor = MaterialTheme.colorScheme.onSurface
                    ),
                    singleLine = true,
                    shape = RoundedCornerShape(16.dp),
                    textStyle = LocalTextStyle.current.copy(fontSize = 13.sp)
                )
            }

            // Status Filter Chips
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 16.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                val filters = listOf("ALL", "ACTIVE", "INACTIVE", "EXPIRED")
                filters.forEach { filter ->
                    val isSelected = viewModel.uiState.statusFilter == filter
                    Surface(
                        modifier = Modifier
                            .weight(1f)
                            .height(36.dp)
                            .clickable { viewModel.onStatusFilterChange(filter) },
                        color = if (isSelected) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.surface,
                        shape = RoundedCornerShape(10.dp),
                        border = if (isSelected) null else androidx.compose.foundation.BorderStroke(
                            1.dp,
                            MaterialTheme.colorScheme.onSurface.copy(alpha = 0.1f)
                        )
                    ) {
                        Box(contentAlignment = Alignment.Center) {
                            Text(
                                text = filter.lowercase().replaceFirstChar { it.uppercase() },
                                color = if (isSelected) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface,
                                fontSize = 11.sp,
                                fontWeight = FontWeight.Bold
                            )
                        }
                    }
                }
            }

            val uiState = viewModel.uiState

            PullToRefreshBox(
                isRefreshing = uiState.isRefreshing,
                onRefresh = { viewModel.refresh() }
            ) {
                if (uiState.filteredMembers.isEmpty() && !uiState.isLoading) {
                    Box(
                        modifier = Modifier.fillMaxWidth().padding(top = 40.dp),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            "No members matching filter",
                            color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.6f),
                            fontSize = 13.sp
                        )
                    }
                } else if (isTablet) {
                    androidx.compose.foundation.lazy.grid.LazyVerticalGrid(
                        columns = androidx.compose.foundation.lazy.grid.GridCells.Fixed(2),
                        modifier = Modifier.fillMaxSize(),
                        horizontalArrangement = Arrangement.spacedBy(16.dp),
                        verticalArrangement = Arrangement.spacedBy(16.dp),
                        contentPadding = PaddingValues(bottom = 24.dp)
                    ) {
                        items(
                            count = uiState.filteredMembers.size,
                            key = { uiState.filteredMembers[it].id }
                        ) { index ->
                            val member = uiState.filteredMembers[index]
                            if (index >= uiState.filteredMembers.size - 1 && uiState.hasMore) {
                                viewModel.loadNextPage()
                            }
                            MemberItem(
                                name = member.name,
                                status = member.status,
                                isActuallyExpired = member.isExpired ?: false,
                                joinDate = member.joiningDateDisplay ?: "N/A",
                                expiryDate = member.expiryDateDisplay ?: "N/A",
                                phone = member.phone ?: "",
                                imageUrl = member.photoUrl,
                                onClick = { onMemberClick(member.id) }
                            )
                        }
                    }
                } else {
                    LazyColumn(
                        modifier = Modifier.fillMaxSize(),
                        verticalArrangement = Arrangement.spacedBy(10.dp),
                        contentPadding = PaddingValues(bottom = 24.dp)
                    ) {
                        items(uiState.filteredMembers.size) { index ->
                            val member = uiState.filteredMembers[index]
                            if (index >= uiState.filteredMembers.size - 1 && uiState.hasMore) {
                                viewModel.loadNextPage()
                            }
                            MemberItem(
                                name = member.name,
                                status = member.status,
                                isActuallyExpired = member.isExpired ?: false,
                                joinDate = member.joiningDateDisplay ?: "N/A",
                                expiryDate = member.expiryDateDisplay ?: "N/A",
                                phone = member.phone,
                                imageUrl = member.photoUrl,
                                onClick = { onMemberClick(member.id) }
                            )
                        }
                    }
                }
            }
        }
    }
}

File: gym_flutter_app/lib/ui/screens/member_list_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../widgets/member_item.dart';
import '../../logic/providers/member_provider.dart';
import 'member_profile_screen.dart';

class MemberListScreen extends ConsumerWidget {
  const MemberListScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final uiState = ref.watch(memberListProvider);
    final theme = Theme.of(context);

    return Scaffold(
      backgroundColor: theme.colorScheme.surface,
      body: LayoutBuilder(
        builder: (context, constraints) {
          final isTablet = constraints.maxWidth > 600;

          return Column(
            children: [
              const SizedBox(height: 60),
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: 16),
                child: Row(
                  children: [
                    IconButton(
                      icon: const Icon(Icons.arrow_back_ios, size: 20),
                      onPressed: () => Navigator.pop(context),
                      color: theme.colorScheme.onSurface,
                    ),
                    Text(
                      "Members",
                      style: theme.textTheme.headlineSmall?.copyWith(
                        fontWeight: FontWeight.w900,
                        color: theme.colorScheme.onSurface,
                        fontSize: isTablet ? 24 : 18,
                      ),
                    ),
                  ],
                ),
              ),
              const SizedBox(height: 16),
              _SearchBar(ref: ref, uiState: uiState),
              const SizedBox(height: 16),
              _FilterChips(ref: ref, uiState: uiState),
              const SizedBox(height: 16),
              Expanded(
                child: RefreshIndicator(
                  onRefresh: () => ref.read(memberListProvider.notifier).refresh(),
                  child: uiState.isLoading && uiState.members.isEmpty
                      ? const Center(child: CircularProgressIndicator())
                      : _MembersList(uiState: uiState, isTablet: isTablet),
                ),
              ),
            ],
          );
        },
      ),
    );
  }
}

class _SearchBar extends StatelessWidget {
  final WidgetRef ref;
  final MemberListUIState uiState;

  const _SearchBar({required this.ref, required this.uiState});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 16),
      child: TextField(
        onChanged: (value) => ref.read(memberListProvider.notifier).onSearchQueryChange(value),
        style: const TextStyle(fontSize: 13),
        decoration: InputDecoration(
          hintText: "Search...",
          hintStyle: TextStyle(color: theme.colorScheme.onSurface.withValues(alpha: 0.5)),
          prefixIcon: Icon(Icons.search, size: 18, color: theme.colorScheme.onSurface.withValues(alpha: 0.5)),
          filled: true,
          fillColor: theme.colorScheme.surface,
          border: OutlineInputBorder(
            borderRadius: BorderRadius.circular(16),
            borderSide: BorderSide.none,
          ),
          contentPadding: const EdgeInsets.symmetric(vertical: 12),
        ),
      ),
    );
  }
}

class _FilterChips extends StatelessWidget {
  final WidgetRef ref;
  final MemberListUIState uiState;

  const _FilterChips({required this.ref, required this.uiState});

  @override
  Widget build(BuildContext context) {
    final filters = ["ALL", "ACTIVE", "INACTIVE", "EXPIRED"];
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 16),
      child: Row(
        children: filters.map((filter) {
          final isSelected = uiState.statusFilter == filter;
          return Expanded(
            child: GestureDetector(
              onTap: () => ref.read(memberListProvider.notifier).onStatusFilterChange(filter),
              child: Container(
                margin: const EdgeInsets.symmetric(horizontal: 4),
                height: 36,
                decoration: BoxDecoration(
                  color: isSelected ? Theme.of(context).colorScheme.primary : Theme.of(context).colorScheme.surface,
                  borderRadius: BorderRadius.circular(10),
                  border: isSelected ? null : Border.all(color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.1)),
                ),
                alignment: Alignment.center,
                child: Text(
                  filter[0].toUpperCase() + filter.substring(1).toLowerCase(),
                  style: TextStyle(
                    color: isSelected ? Colors.black : Theme.of(context).colorScheme.onSurface,
                    fontSize: 11,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ),
            ),
          );
        }).toList(),
      ),
    );
  }
}

class _MembersList extends StatelessWidget {
  final MemberListUIState uiState;
  final bool isTablet;

  const _MembersList({required this.uiState, required this.isTablet});

  @override
  Widget build(BuildContext context) {
    if (uiState.filteredMembers.isEmpty) {
      return Center(
        child: Text(
          "No members matching filter",
          style: TextStyle(color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.6), fontSize: 13),
        ),
      );
    }

    if (isTablet) {
      return GridView.builder(
        padding: const EdgeInsets.fromLTRB(16, 0, 16, 24),
        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
          crossAxisCount: 2,
          crossAxisSpacing: 16,
          mainAxisSpacing: 16,
          mainAxisExtent: 140, // Height of MemberItem roughly
        ),
        itemCount: uiState.filteredMembers.length,
        itemBuilder: (context, index) {
          final member = uiState.filteredMembers[index];
          return MemberItem(
            name: member.name,
            status: member.status,
            isActuallyExpired: member.isExpired,
            joinDate: member.joiningDateDisplay,
            expiryDate: member.expiryDateDisplay,
            phone: member.phone,
            imageUrl: member.photoUrl,
            onClick: () {
              Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => MemberProfileScreen(memberId: member.id)),
              );
            },
          );
        },
      );
    }

    return ListView.separated(
      padding: const EdgeInsets.fromLTRB(16, 0, 16, 24),
      itemCount: uiState.filteredMembers.length,
      separatorBuilder: (context, index) => const SizedBox(height: 10),
      itemBuilder: (context, index) {
        final member = uiState.filteredMembers[index];
        return MemberItem(
          name: member.name,
          status: member.status,
          isActuallyExpired: member.isExpired,
          joinDate: member.joiningDateDisplay,
          expiryDate: member.expiryDateDisplay,
          phone: member.phone,
          imageUrl: member.photoUrl,
          onClick: () {
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => MemberProfileScreen(memberId: member.id)),
            );
          },
        );
      },
    );
  }
}
